sherlockml/clients/auth.py

```diff
@@ -46,3 +47,54 @@ def get_access_token(self, client_id, client_secret):
         expires_at = now + timedelta(seconds=body['expires_in'])
 
         return AccessToken(token, expires_at)
+
+
+class SherlockMLAuth(AuthBase):
+    """Requests auth implementation for accessing SherlockML services.
+
+    Parameters
+    ----------
+    auth_service_url : str
+        The URL of the SherlockML authentication service
+    client_id : str
+        The SherlockML client ID to use for authentication
+    client_secret : str
+        The client secret associated with the client ID
+
+    To perform an authenticated request against a SherlockML service, first
+    construct an instance of this class:
+
+    >>> auth = SherlockMLAuth('https://hudson.example.sherlockml.net',
+                              your_client_id, your_client_secret)
+
+    then pass it as the ``auth`` argument when constructing a ``requests``
```

Comment:
Is it useful to use the Sphinx reference notation for this?
```
:class:`requests.Session`
```

Reply:
Have implemented this.